Tasks/AzurePowerShellV2/AzurePowerShell.ps1

```diff
@@ -0,0 +1,147 @@
+Trace-VstsEnteringInvocation $MyInvocation
+Import-VstsLocStrings "$PSScriptRoot\Task.json"
+
+# Get inputs.
+$scriptType = Get-VstsInput -Name ScriptType -Require
+$scriptPath = Get-VstsInput -Name ScriptPath
+$scriptInline = Get-VstsInput -Name Inline
+$scriptArguments = Get-VstsInput -Name ScriptArguments
+$targetAzurePs = Get-VstsInput -Name TargetAzurePs
+$customTargetAzurePs = Get-VstsInput -Name CustomTargetAzurePs
+
+# Validate the script path and args do not contains new-lines. Otherwise, it will
+# break invoking the script via Invoke-Expression.
+if ($scriptType -eq "FilePath") {
+    if ($scriptPath -match '[\r\n]' -or [string]::IsNullOrWhitespace($scriptPath)) {
+        throw (Get-VstsLocString -Key InvalidScriptPath0 -ArgumentList $scriptPath)
+    }
+}
+
+if ($scriptArguments -match '[\r\n]') {
+    throw (Get-VstsLocString -Key InvalidScriptArguments0 -ArgumentList $scriptArguments)
+}
+
+# string constants
+$otherVersion = "OtherVersion"
+$latestVersion = "LatestVersion"
+
+if ($targetAzurePs -eq $otherVersion) {
+    if ($customTargetAzurePs -eq $null) {
+        throw (Get-VstsLocString -Key InvalidAzurePsVersion $customTargetAzurePs)
+    } else {
+        $targetAzurePs = $customTargetAzurePs.Trim()        
+    }
+}
+
+$pattern = "^[0-9]+\.[0-9]+\.[0-9]+$"
+$regex = New-Object -TypeName System.Text.RegularExpressions.Regex -ArgumentList $pattern
+
+if ($targetAzurePs -eq $latestVersion) {
+    $targetAzurePs = ""
+} elseif (-not($regex.IsMatch($targetAzurePs))) {
+    throw (Get-VstsLocString -Key InvalidAzurePsVersion -ArgumentList $targetAzurePs)
+}
+
+. "$PSScriptRoot\Utility.ps1"
+$targetAzurePs = Get-RollForwardVersion -azurePowerShellVersion $targetAzurePs
+
+$authScheme = ''
+try
+{
+    $serviceNameInput = Get-VstsInput -Name ConnectedServiceNameSelector -Default 'ConnectedServiceName'
+    $serviceName = Get-VstsInput -Name $serviceNameInput -Default (Get-VstsInput -Name DeploymentEnvironmentName)
+    if (!$serviceName)
+    {
+            Get-VstsInput -Name $serviceNameInput -Require
+    }
+
+    $endpoint = Get-VstsEndpoint -Name $serviceName -Require
+
+    if($endpoint)
+    {
+        $authScheme = $endpoint.Auth.Scheme
+    }
+
+    Write-Verbose "AuthScheme $authScheme" 
+}
+catch
+{
+   $error = $_.Exception.Message
+   Write-Verbose "Unable to get the authScheme $error" 
+}
+
+if ($endpoint.Auth.parameters.authenticationType -ieq "spnCertificate")
+{
+	throw (Get-VstsLocString -Key CertificateEndpointNotSupported)
+}
```

Comment:
We dont need this. AzurePowerShellV2 will automatically start using the latest common module which has support for certificate based SPN.